# Proyecto modelo de ahorro para un viaje familiar

- Cecilia Alejandra Mancilla Olivera
- Jesus Emmanuel Palomera Gaytan
- Hugo Gabriel Velarde Pérez

### Planteamiento del problema

En este proyecto se creará un ahorro de $150,000 para un viaje familiar, en donde el dinero se invertirá en un banco que ofrece una tasa de interés mensual del 2.4%.

Se planea realizar un viaje familiar de 6 personas, que quieren visitar Suecia, la duración del viaje será de 2 semanas y cuentan con un capital inicial de $\$50,000$ pesos para empezar a realizar el ahorro.
La cantidad que quieren ahorrar es de $\$150,000$, sin inversiones adicionales. ¿Cuánto tiempo se debe de dejar invertido el dinero?

La familia planea invertir todo su dinero a un fondo de inversión para así disponer de más capital para su viaje.
El Banco "Fulanito de tal" les ofrece una tasa de interés del $2.4\%$ mensual.

# Objetivos

### Objetivo principal
- El objetivo **principal** será ver si es posible que la familía se vaya de viaje con un capital inicial de $ 50,000 $ pesos y proponer alguna alternativa de ahorro en caso de no ser posible solo invirtiendo en el fondo de inversión.

##### Objetivos secundarios
- El **primer** objetivo secundario es poder obtener la cantidad de tiempo en meses que se debería de dejar invertido el dinero para que se vayan de viaje.
- El **segundo** objetivo secundario es ver por cuantos meses tendrían que ahorrar y además invertir en el fondo para poder lograr irse en el tiempo deseado.

In [1]:
import numpy as np
import matplotlib.pyplot as plt
from scipy.integrate import odeint
from ipywidgets import *
%matplotlib inline

### Modelo que representa el problema.
-Condiciones iniciales de acuerdo al problema 

In [2]:
C0  = 50000
i   = 0.024
Ck  = 150000

k   = np.log(Ck/C0)/np.log(1+i)
k   = np.ceil(k)
k

47.0

En caso de que la familia solo inviertan en el fondo de inversión, estarían tardando un periodo de 47 meses. 

### Visualización de la solución del problema

In [3]:
C  = C0*(1+i)**k
C

152429.1284333982

En este periodo de 47 meses lograrían ahorrar una cantidad de 152429.1284333982.

Si graficamos los datos conforme el tiempo y el capital, el periodo a inversión luciría como se muestra a continuación.

In [4]:
def cap(C0, r, Ct=150000) :
    t = np.log(Ct/C0)/r
    T = np.linspace(0, t, 1000)
    def fun(C, t, r) :
        return r*C
    C = odeint(fun, C0, T, args=(r,))[:, 0]
    
    plt.figure()
    plt.plot(T, C)
    plt.grid()
    plt.xlabel("meses")
    plt.ylabel("MXN")
    

i  = 0.024
C0 = 50000
r  = np.log(1+i)
interact(cap, C0=fixed(50000), r=fixed(r), Ct=(50000, 150000, 10000))

interactive(children=(IntSlider(value=150000, description='Ct', max=150000, min=50000, step=10000), Output()),…

<function __main__.cap(C0, r, Ct=150000)>

Si la familia quiere irse de vacaciones antes de los 3 años y 11 meses, y planea viajar en 2 años ¿cuánto dinero debería ahorrar mensualmente?, teniendo en cuenta que su capital inicial no ha incrementado. 

In [5]:
viaje   = 150000
C0      = 50000
periodo = 2
i       = 0.024
n       = periodo*12

In [6]:
import numpy_financial as npf

In [7]:
A = -npf.pmt(-i, n, viaje-C0)
A

3032.4138539548467

La familia tendría que ahorrar, 3033 pesos por mes durante 24 meses para poder llegar a su meta de ahorro y además dejar invertido su capital inicial en el fondo de inversión original para generar el dinero faltante.

## Solución del problema

In [8]:
3032.4138539548467*24+C0*(1+.024)**24

161120.28573383557

Con la inversión en el fondo y el ahorro al final de los 2 años tendrían 161120.30 pesos

In [9]:
3032.4138539548467*24+C0-viaje

-27222.067505083687

27,222.07 sería el dinero que faltaria después del ahorro y el cuál obtendriamos de la inversión.

## Conclusiones

- Se puede concluir que si la familia desea viajar y de ser asi acelerar su viaje podrian ahorrar 3033 pesos mensuales aparte de su inversion y lograr su viaje en 2 años en lugar de esperar 3 años y 11 meses
- 